webapp/src/pages/Settings.tsx

```diff
@@ -31,48 +82,625 @@ const Settings: React.FC = () => {
+      sx={{ paddingTop: 0.5 }}
+      size="small"
+      checked={selected}
       disabled={isError || isFetching}
-      onChange={(_, checked) =>
-        setPartialConfig({ ...partialConfig, [field]: checked ? {} : null })
+      onChange={(...[, checked]) =>
+        setPartialConfig({
+          ...partialConfig,
+          [field]: checked ? data?.[field] ?? CONFIG_SUB_FIELDS[field] : null,
+        })
       }
     />
   );
 
-  return (
-    <Container maxWidth="sm">
-      <FormControl sx={{ width: "100%" }}>
-        <FormGroup>
-          <FormControlLabel
-            control={switchNullOrDefault("behavioral_testing")}
-            label="Perturbation testing"
-          />
+  const displayArgumentsList = (
+    name: string,
+    argEntries: Record<string, string> | Record<string, string>[]
+  ) => (
+    <Box display="flex" flexDirection="column" paddingTop={1}>
+      <Typography variant="caption">{name}</Typography>
+      {Object.entries(argEntries).map(([field, value], index) => (
+        <Box
+          key={index}
+          display="flex"
+          flexDirection="row"
+          justifyContent="flex-start"
+        >
+          <Typography
+            variant="body2"
+            sx={{
+              s: 1,
+              width: "100%",
+            }}
+          >
+            {field}:
+          </Typography>
+          <Tooltip title={value} placement="bottom">
+            <Typography
+              variant="body2"
+              sx={{
+                s: 1,
+                width: "20ch",
+                whiteSpace: "nowrap",
+                overflow: "hidden",
+                textOverflow: "ellipsis",
+              }}
+            >
+              {value}
+            </Typography>
+          </Tooltip>
+        </Box>
+      ))}
+    </Box>
+  );
+
+  const displayReadonlyFields = (label: string, value: string | undefined) => (
+    <TextField
+      sx={{ m: 1, width: "40ch" }}
+      size="small"
+      variant="standard"
+      label={<Typography fontWeight="bold">{label}</Typography>}
+      value={value}
+      disabled={isError || isFetching}
+      InputProps={{
+        readOnly: true,
+        disableUnderline: true,
+        style: { fontSize: 14 },
+      }}
+    />
+  );
+
+  const displayNumberField = (
+    config: keyof AzimuthConfig,
+    field: string,
+    value: number | undefined
+  ) => (
+    <TextField
+      id={field}
+      size="small"
+      label={<Typography fontWeight="bold">{field}</Typography>}
+      type="number"
+      value={value}
+      disabled={!Boolean(resultingConfig[config])}
+      variant="standard"
+      onChange={(event) =>
+        setPartialConfig(
+          _.merge({}, partialConfig, {
+            [config]: _.merge(
+              { [field]: Number(event.target.value) },
+              config in partialConfig
+                ? _.get(partialConfig, config)
+                : _.get(resultingConfig, config)
+            ),
```

Comment:
This doesn't work at the moment. The number pickers in the Analyses Customization are not editable. Multiple things:
1. Is there a reason for using `_.get(resultingConfig, config)` instead of native `resultingConfig[config]`?
2. About the ternary operator: if something is `in partialConfig`, it will automatically be in `resultingConfig`. That's the purpose of `resultingConfig`: as a merge of `partialConfig` over `data`. So, this is equivalent to
```diff
-              config in partialConfig
-                ? _.get(partialConfig, config)
-                : _.get(resultingConfig, config)
+              resultingConfig.config
```
3. The order in your `_.merge` is important. You are currently overwriting the new value with `resultingConfig.config`, so it never takes effect. They should be merged in inverse order. In that case, as you have correctly done elsewhere, you need to pass `{}` first to avoid modifying the `resultingConfig` in place.
4. A `_.assign` would be sufficient here, as we don't need a recursive `_.merge`.
```suggestion
            [config]: _.assign({}, resultingConfig[config], {
              [field]: Number(event.target.value),
            }),
```